src/lib.rs

```diff
@@ -91,3 +92,95 @@ pub fn format_size(size: u64, binary: bool) -> String {
         NumberPrefix::Prefixed(prefix, number) => format!("{:.2} {}B", number, prefix),
     }
 }
+
+/// Calculate the sum of sizes of all entries
+///
+/// Does not account twice the size of entries that are within other entries.
+/// Implementation uses BTreeMap to build a path Trie.
+pub fn calculate_unique_total_size(entries: &[Entry]) -> u64 {
+    let mut filtered_entries = Vec::<&Entry>::new();
+    let mut canonicalized_paths = Vec::<PathBuf>::new();
+
+    // Canonicalize each path, silently ignoring failures.
+    // TODO: Review if we should ignore failures.
```

Comment:
What should happen in here?

Reply:
The [documentation](https://doc.rust-lang.org/std/fs/fn.canonicalize.html) says

> This function will return an error in the following situations, but is not limited to just these cases:
> - `path` does not exist.
> - A non-final component in path is not a directory.

Since `Entry`s already have their size, we can't safely assume that both of these have been checked.

However, since it "is not limited to just these cases", it might be safer to at least warn the user if something else goes wrong (especially since we support multiple platforms, and one of them is not particularly trustworthy :eyes:).

I therefore suggest using the `log_err` method on that `Result`, e.g.:
```diff
    for entry in entries {
+        let canonical_path = entry.path.canonicalize().log_err(Some(&entry.path));
+        if let Ok(path) = canonical_path {
-        if let Ok(path) = entry.path.canonicalize() {
            filtered_entries.push(entry);
            canonicalized_paths.push(path);
        }
    }
```